Handling JSON in MongoDB (#2791)
section-engineering-education/engineering-education: content/articles/handling-json-in-mongodb/index.md

> ### Introduction
> MongoDB is a high-performance, document-oriented NoSQL database which came to light in the mid-2000s. Unlike SQL-based databases, MongoDB uses documents (records) and collections (tables) to store non-relational data. MongoDB has grown in its use due to the scalability and its ease of use.

Reviewer: ```suggestion
---
layout: engineering-education
status: publish
published: true
url: /handling-json-in-mongodb/
title: Handling JSON in MongoDB
description: In this article, we will learn how to structure, import, export, and save JSON data in MongoDB.
author: simon-salva
date: 2021-07-13T00:00:00-00:20
topics: []
excerpt_separator: <!--more-->
images:
  - url: /engineering-education/handling-json-in-mongodb/hero.png
    alt: Handling JSON in MongoDB image
---

MongoDB is a high-performance, document-oriented NoSQL database which came to light in the mid-2000s. Unlike SQL-based databases, MongoDB uses documents (records) and collections (tables) to store non-relational data.
<!--more-->
MongoDB has grown in its use due to the scalability and its ease of use.
```
Added front matter.

```json
[
    {
        "student_number": 1,
        "name": "John Doe",
        "age": 17,
        "Course": "Computer Science"
    },
    {
        "student_number": 2,
        "name": "Joan Dee",
        "age": 23,
        "Course": "Computer Animation"
    },
    {
        "student_number": 3,
        "name": "Jim Doeh",
        "age": 19,
        "Course": "Industrial Science"
    },
    {
        "student_number": 4,
        "name": "John Dohn",
        "age": 37,
        "Course": "Political Science"
    }
]
```

### Import JSON to MongoDB
Next, you need to create a database to hold the JSON object that you will import. Execute the command below to create the database and import the JSON file as a `student` collection.
```bash
mongoimport --jsonArray --db studentinfo --collection students --file path-to-your-file\filename.json
```

If your import was a success, you should get the message below in the terminal
```bash
2021-06-29T22:35:18.454+0300    connected to: mongodb://localhost/
2021-06-29T22:35:18.500+0300    4 document(s) imported successfully. 0 document(s) failed to import.
```

Now to verify the import, run the command below to view the imported data:

> ```bash
> db.students.find().pretty()
> ```

Reviewer: ```suggestion
```bash
db.students.find()
```
```

WriteResult({ "nInserted" : 1 })
```

Now if we run the command, we find an extra record in the database:
```json
db.students.find()
{ "_id" : ObjectId("60db758c4a433597bcae61c0"), "student_number" : 1, "name" : "John Doe", "age" : 17, "Course" : "Computer Science" }
{ "_id" : ObjectId("60db758c4a433597bcae61c1"), "student_number" : 4, "name" : "John Dohn", "age" : 37, "Course" : "Political Science" }
{ "_id" : ObjectId("60db758c4a433597bcae61c2"), "student_number" : 2, "name" : "Joan Dee", "age" : 23, "Course" : "Computer Animation" }
{ "_id" : ObjectId("60db758c4a433597bcae61c3"), "student_number" : 3, "name" : "Jim Doeh", "age" : 19, "Course" : "Industrial Science" }
{ "_id" : ObjectId("60db7a59ee65088f9b51677e"), "student_number" : 5, "name" : "Last student added", "age" : 22, "Course" : "Engineering" }
```

### Export JSON to MongoDB Database
Now let us export the new file to JSON using the following command:
```shell
mongoexport --db studentinfo --collection student --out d:\out.json --pretty
```

You should have the response below in your terminal:
```shell
2021-06-29T23:02:50.046+0300    connected to: mongodb://localhost/
2021-06-29T23:02:50.071+0300    exported 4 records

> ```

Reviewer: Did you forget to change these ones? Please change `shell` to `bash`.

Now if you navigate to your D drive, you should find your exported JSON file named `out.json`, and the records are as below:
```json
{
    "_id": {
        "$oid": "60db75f6f8992a3dd20bb089"
    },
    "student_number": 3,
    "name": "Jim Doeh",
    "age": 19,
    "Course": "Industrial Science"
}
{
    "_id": {
        "$oid": "60db75f6f8992a3dd20bb08a"
    },
    "student_number": 2,
    "name": "Joan Dee",
    "age": 23,
    "Course": "Computer Animation"
}
{
    "_id": {
        "$oid": "60db75f6f8992a3dd20bb08b"
    },
    "student_number": 1,
    "name": "John Doe",
    "age": 17,
    "Course": "Computer Science"
}
{
    "_id": {
        "$oid": "60db75f6f8992a3dd20bb08c"
    },
    "student_number": 4,
    "name": "John Dohn",
    "age": 37,
    "Course": "Political Science"
}
```

### Conclusion
In this article, we've learned how to handle JSON data in MongoDB and the various operations that can be performed on JSON using MongoDB. Among the operations are structuring data, importing and exporting data from JSON to MongoDB, and vice versa.
You can find the JSON files [here.](https://github.com/salvador02/jsonfile)

> ### Further Reading
> Now that you have an understanding of how to use JSON in MongoDB, you can check out these links to find out more on this topic:
> - [JSON and BSON](https://www.mongodb.com/json-and-bson)
> - [Mongoexport](https://docs.mongodb.com/manual/reference/program/mongoexport/)
> - [Mongoimport](https://docs.mongodb.com/manual/reference/program/mongoimport/)

Reviewer: ```suggestion
### Further Reading
Now that you have an understanding of how to use JSON in MongoDB, you can check out these links to find out more on this topic:
- [JSON and BSON](https://www.mongodb.com/json-and-bson)
- [Mongoexport](https://docs.mongodb.com/manual/reference/program/mongoexport/)
- [Mongoimport](https://docs.mongodb.com/manual/reference/program/mongoimport/)

Happy coding!

---

Peer Review Contributions by: [Eric gacoki](/engineering-education/authors/eric-gacoki/)
```
Added peer reviewer line.

### Introduction
MongoDB is a high-performance, document-oriented NoSQL database which came to light in the mid-2000s.Unlike SQL-based databases, MongoDB uses documents (records) and collections (tables) to store non-relational data. MongoDB database has grown in its use because of its scalability and ease of use.

JavaScript Object Notation (JSON) is a data exchange format solely based on JavaScript. JSON objects can easily get transferred from one system to another due to their compatibility with the majority of systems. JSON is simple and easy to use besides being independent of any language. For these reasons, JSON has a wide range of applications in programming and data representation.

JSON can store various data types like strings, arrays, objects, and Binary JSON (BSON). What makes access to these data easy is the ability of various tools to export and import JSON documents to and from other systems using the various available tools as JSON makes use of key-value pair notation presumed to be the basic units of data.

> ### Table of Contents

Reviewer: ```suggestion
### Table of content
```
Only capitalize the first word not multiple words in the heading, unless it's a proper noun, an acronym or a keyword.

```json
[
    {
        "student_number": 1,
        "name": "John Doe",
        "age": 17,
        "Course": "Computer Science"
    },
    {
        "student_number": 2,
        "name": "Joan Dee",
        "age": 23,
        "Course": "Computer Animation"
    },
    {
        "student_number": 3,
        "name": "Jim Doeh",
        "age": 19,
        "Course": "Industrial Science"
    },
    {
        "student_number": 4,
        "name": "John Dohn",
        "age": 37,
        "Course": "Political Science"
    }
]
```

### Import JSON to MongoDB
Next, you need to create a database to hold the JSON object that you will import. Execute the command below to create the database and import the JSON file as a `student` collection.
```bash
mongoimport --jsonArray --db studentinfo --collection students --file path-to-your-file\filename.json
```

If your import was a success, you should get the message below in the terminal
```bash
2021-06-29T22:35:18.454+0300    connected to: mongodb://localhost/
2021-06-29T22:35:18.500+0300    4 document(s) imported successfully. 0 document(s) failed to import.
```

Now to verify the import, run the command below to view the imported data:
```bash

> db.students.find()

Reviewer: @salvador02,
Seems like I mistakenly removed the closing back-ticks (```) in my previous suggestion. Kindly fix this and the article will be ready for `final review` 😎.
